packages/frontend/src/models/ModelInfoUI.ts

```diff
@@ -0,0 +1,23 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { ModelInfo } from '@shared/src/models/IModelInfo';
+
+export interface ModelInfoUI extends ModelInfo {
+  actionError?: string;
```

Comment:
I am not sure to understand why you created an `actionError` here ? The error is not stored in the Task ?

Reply:
Yes, it is taken from there. By using the object we can keep the logic in the Models and just pass the result to the ModelColumnActions to display it. Otherwise we have to subscribe to the tasks inside the modelcolumnactions

Reply:
I think we should not create a Modelinfo, as we consider the file corrupted if the sha is not matching. Therefore it should stop at the Download level, in a similar maner than a connection RESET or something failing the download task

Reply:
I mean, I think we do not need to have this error icon on the model row. I mean, when we have an error, we could keep the error at the top.

When downloading a recipe, if we put the error in the task the user will see it, so we could just have the same for the the models page?

Reply:
Ah ok, got it. I'll give it a try